specs/core/0_beacon-chain.md

```diff
@@ -946,27 +946,26 @@ def get_crosslink_committees_at_slot(state: BeaconState,
     """
     Returns the list of ``(committee, shard)`` tuples for the ``slot``.
     """
-    state_epoch_slot = state.slot - (state.slot % EPOCH_LENGTH) 
-    assert state_epoch_slot <= slot + EPOCH_LENGTH
-    assert slot < state_epoch_slot + EPOCH_LENGTH
-    offset = slot % EPOCH_LENGTH
+    assert slot >= state.previous_epoch_calculation_slot
 
-    if slot < state_epoch_slot:
+    if slot < state.current_epoch_calculation_slot:
         committees_per_slot = get_previous_epoch_committee_count_per_slot(state)
```

Comment:
I'm pretty sure this is wrong, and `state_epoch_slot` is correct. The `current_epoch_calculation_slot` is the slot the data from which is used to calculate the current shuffling, which could be prior to the current slot.

Reply:
Right and if it is prior, we fall through to the `else` and use the `current_` seed, calc slot, etc.

I think this is essentially doing to same thing but more explicitly uses the slots we stored in state and I think is clearer.

The future slots is not generally safe so I'll add the restriction back in.